cli/compose/types/types.go

```diff
@@ -260,6 +262,15 @@ type ServiceSecretConfig struct {
 	Mode   *uint32
 }
 
+// ServiceConfigObjConfig is the config obj configuration for a service
+type ServiceConfigObjConfig struct {
```

Comment:
Could we use this instead these actually diverge:

```go
type ServiceConfigObjConfig ServiceSecretConfig
```

or 

```
type ServiceConfigObjConfig struct {
 ServiceSecretConfig
}
```
That might let us re-use some code as well? 

And we could leave a comment about changing them when we need to